api/src/main/java/org/bahmni/module/fhircdss/api/service/impl/ConditionsRequestBuilder.java

```diff
@@ -159,4 +162,22 @@ private Obs getObsFor(Obs visitDiagnosisObsGroup, String conceptName) {
         }
         return null;
     }
+
+    private void updateConditionDisplayNameToShortName(Condition conditionEntry) {
+        Optional<Coding> optionalCoding = conditionEntry.getCode().getCoding().stream().filter(coding -> coding.getSystem() == null && coding.getCode() != null).findFirst();
+        if (optionalCoding.isPresent()) {
+            Coding coding = optionalCoding.get();
+            String conceptUuid = coding.getCode();
```

Comment:
shouldn't you also match the system to be of openmrs or a given codesystem?

Reply:
yes @angshu .. matching against openmrs for fetching concept uuid. 
system is not available for openmrs in OpenMRS FHIR Condition resource.
```
"resource": {
                "resourceType": "Condition",
                "id": "311a8c68-cdff-4aab-87c5-d4def0e24dcb",
                "meta": {
                    "lastUpdated": "2023-06-09T05:07:13.000+00:00"
                },
                "text": {
                    ....
                },
                "clinicalStatus": {
                    "coding": [
                        ...
                    ]
                },
                "code": {
                    "coding": [
                        {
                            "code": "fdf78c69-2907-47b9-bc11-119f6170f454",
                            "display": "Large liver (disorder)"
                        },
                        {
                            "system": "http://snomed.info/sct",
                            "code": "xxxxxxx",
                            "display": "Large liver (disorder)"
                        }
                    ],
                    "text": "Large liver (disorder)"
                },
                "subject": {
                    ...
                },
                "recordedDate": "2023-06-09T05:07:13+00:00",
                "recorder": {
                    ...
                }
            }
```

Reply:
then should the clause be that if (coding.system = empty / null), then only try to get concept with coding.code

Reply:
yes @angshu, doing the same filter in the line # 167